### ISSUE TITLE
workspace.json with version 2 incompatible error

### ISSUE THREAD

**Issue by nevraydin** [nrwl/nx#8018]
## Current Behavior

`>  npm start --verbose`
```
> my-workspace@0.0.0 start
> nx serve


> nx run framework-mf-container:serve:development 
'project.json' files are incompatible with version 1 workspace schemas.
```

———————————————————————————————————————————————

>  NX   ERROR  Running target "framework-mf-container:serve" failed

## Steps to Reproduce
Just trying arrange workspace.json file when upgrade my nx workspace with nx 13. When adding more projects, I got an error like below. Is there any limit or something for this file?

**'project.json' files are incompatible with version 1 workspace schemas.**

### Failure Logs

```
Error: 'project.json' files are incompatible with version 1 workspace schemas.
    at ..\node_modules\@nrwl\tao\src\shared\workspace.js:228:19
    at Array.forEach (<anonymous>)
    at toOldFormatOrNull (..\node_modules\@nrwl\tao\src\shared\workspace.js:225:37)
    at MapSubscriber.project (..\@nrwl\tao\src\commands\ngcli-adapter.js:145:81)
    at MapSubscriber._next (..\my-workspace\node_modules\rxjs\internal\operators\map.js:49:35)
    at MapSubscriber._next (..\node_modules\rxjs\internal\operators\map.js:55:26)
    at DefaultIfEmptySubscriber._next (..\node_modules\rxjs\internal\operators\defaultIfEmpty.js:41:26)
```

### Environment

```
workspace.json
{
 "version": 2,
  "projects": {
    "framework-mf-container": "apps/framework/mf-container"
 }
}

```
```
    Node: v16.13.0
    nx: 13.2.0-rc.4
    npm 8.1.4

    "@nrwl/cli": "13.2.0-rc.4",
    "@nrwl/cypress": "13.2.0-rc.4",
    "@nrwl/devkit": "13.2.0-rc.4",
    "@nrwl/eslint-plugin-nx": "13.2.0-rc.4",
    "@nrwl/jest": "13.2.0-rc.4",
    "@nrwl/linter": "13.2.0-rc.4",
    "@nrwl/nx-cloud": "12.5.4",
    "@nrwl/react": "13.2.0-rc.4",
    "@nrwl/tao": "13.2.0-rc.4",
    "@nrwl/web": "13.2.0-rc.4",
    "@nrwl/workspace": "13.2.0-rc.4",
    "@testing-library/react": "11.2.6",
    "@types/jest": "27.0.2",
    "@types/marked": "^0.7.4",
    "@types/node": "14.14.33",
    "@types/react": "17.0.3",
    "@types/react-dom": "17.0.3",
    "@types/react-router-dom": "5.1.7",
    "@types/recordrtc": "^5.6.2",
    "@typescript-eslint/eslint-plugin": "4.33.0",
    "@typescript-eslint/parser": "4.33.0",
    "codelyzer": "~5.0.1",
    "cypress": "^8.3.0",
    "dotenv": "10.0.0",
    "eslint": "7.32.0",
    "eslint-config-prettier": "8.1.0",
    "eslint-plugin-cypress": "^2.10.3",
    "eslint-plugin-import": "2.22.1",
    "eslint-plugin-jsx-a11y": "6.4.1",
    "eslint-plugin-react": "7.23.1",
    "eslint-plugin-react-hooks": "4.2.0"
```

**Comment by AgentEnder** [comment]
There's not a limit on workspace.json, so long as that version flag is set to 2 and such.

Can you link a reproduction repository? I've not been able to replicate the failure.

**Comment by nevraydin** [comment]
I logged `resolveInlineProjectConfigurations `function in `ngcli-adapter`.js. When reading `workspace.json`, after the app key, it continues as string. It is only happen 500-600+ projects in `workspace.json`.

Github link: https://github.com/nevraydin/nx-examples/


```
resolveInlineProjectConfigurations(config) {
        var _a;
        // Creates an observable where each emission is a project configuration
        // that is not listed inside workspace.json. Each time it encounters a
        // standalone config, observable is updated by concatenating the new
        // config read operation.
        let observable = rxjs_1.EMPTY;
        console.log(config);
        Object.entries((_a = config.projects) !== null && _a !== void 0 ? _a : {}).forEach(([project, projectConfig]) => {
            if (typeof projectConfig === 'string') {
                // configFilePath is not written to files, but is stored on the config object
                // so that we know where to save the project's configuration if it was updated
                // by another angular schematic.
                const configFilePath = (0, path_1.join)(projectConfig, 'project.json');
                const next = this.read(configFilePath).pipe((0, operators_1.map)((x) => ({
                    project,
                    projectConfig: Object.assign(Object.assign({}, (0, json_1.parseJson)(Buffer.from(x).toString())), { configFilePath }),
                })));
                observable = observable ? (0, rxjs_1.concat)(observable, next) : next;
            }
        });
        return observable.pipe(
        // Collect all values from the project.json read operations
        (0, operators_1.toArray)(), 
        // Use these collected values to update the inline configurations
        (0, operators_1.map)((x) => {
            x.forEach(({ project, projectConfig }) => {
                config.projects[project] = projectConfig;
            });
            return config;
        }));
    }
}
```

Output 
![image](https://user-images.githubusercontent.com/946627/145024844-eadf656a-017a-4def-9e50-891293809b75.png)

**Comment by AgentEnder** [comment]
Gotcha, this is a known issue with greater than about 600 projects. It was fixed in #8026 yesterday 😃.

This will be included in 13.3's release.

**Comment by nevraydin** [comment]
Thank you. 😁👍🏻
